docs/src/views/CustomComponents.tsx

```diff
@@ -0,0 +1,234 @@
+                root: ({ props, variables, theme: { siteVariables } }) => ({
+                  backgroundColor: siteVariables.brand,
+                  color: (variables as any).color,
+                }),
+              },
+            },
+            componentVariables: {
+              StyledButton: {
+                color: '#F2F2F2',
+              },
+            },
+          }}
+        >
+          <StyledButton>Provider styled button</StyledButton>
+        </Provider>
+      )}
+    />
+
+    <p>
+      The second argument of the <code>createComponent</code> config param is the render method.
+      This is the place where you link the Stardust bits with your custom component. This method is
+      invoked with the following parameters:
+    </p>
+    <ul>
+      <li>
+        <code>stardust</code> - this is the object containing the evaluated theming props (classes
+        and rtl)
+      </li>
+      <li>
+        <code>...props</code> - all other props provided by the user
+      </li>
+    </ul>
+    <p>
+      The previous implementation of the render method inside the <code>createComponent</code>{' '}
+      function for the
+      <code>StyledButton</code>, allows the user, to use the styles and variables properties on the
+      custom component in the same way they would be used for any Stardust component.
+    </p>
+
+    <Header as="h2" content="Using the custom components" />
+
+    <p>
+      We already saw how the <code>Provider</code> can define some stylings and variables for the
+      custom components. Next, we will take a look into few examples of how the user can further
+      customize the styling and variables of the custom components, the same way they would do with
+      the Stardust components.
+    </p>
+
+    <Header as="h3" content="Example 1. Using the variables property" />
+
+    <ExampleSnippet
+      value={[
+        `<StyledButton variables={(siteVars) => ({color: siteVars.black})}>`,
+        `  Inline styled button`,
+        `</StyledButton>`,
+      ].join('\n')}
+      render={() => (
+        <Provider
+          theme={{
+            componentStyles: {
+              StyledButton: {
+                root: ({ props, variables, theme: { siteVariables } }) => ({
+                  backgroundColor: siteVariables.brand,
+                  color: (variables as any).color,
+                }),
+              },
+            },
+            componentVariables: {
+              StyledButton: {
+                color: '#F2F2F2',
+              },
+            },
+          }}
+        >
+          <StyledButton variables={siteVariables => ({ color: siteVariables.black })}>
+            inline styled button
+          </StyledButton>
+        </Provider>
+      )}
+    />
+
+    <Header as="h3" content="Example 2. Using the styles property" />
+
+    <ExampleSnippet
+      value={[
+        `<StyledButton styles={({theme: {siteVariables}}) => ({backgroundColor: siteVariables.black})}>`,
```

Comment:
I would rather strive for simplifying examples as much as possible. We should just show that it is _possible_ to use styles prop, as well as suggest the fact that it could be used the same way as for Stardust components. Then I see us providing simple example that might prove that, and after this example make a sentence that **all other advanced scenarios are supported as well, please, take a look on the corresponding [docs](#)**. 

This will allow us
-  to simplify example's code (crucially important thing)
- not introduce additional maintainability burden to us - say, in case if implementation of `styles` prop change, we won't need to fix this example's code